### PR TITLE
Clean up RpcServer

### LINKED ISSUE #144: Use base64 lib included in RpcServer for cookie decoding?

### PR DESCRIPTION
**Pull request by sisou**
Extracts `JsonUtils`, removes `Base64`, add proper typings and linting, removes need for RpcServer.d.ts definition file.

Resolves #144.

### CODE REVIEW

**Comment by sisou** [comment]
Thanks, fixed it.

File: src/lib/JsonUtils.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+     * @returns {any}
+     */
+    static _jsonifyType(key, value) {
+        if (value instanceof Uint8Array) {
+            return JsonUtils._typedObject(JsonUtils.ExtraJsonTypes.UINT8_ARRAY, Nimiq.BufferUtils.toBase64(value));
+        }
+        return value;
+    }
+
+    /* eslint-disable-next-line valid-jsdoc */
```
Comment (nibhar) [inline src/lib/JsonUtils.js]: Is unnecessary I think.

File: src/lib/RpcServer.es.js [hunk 2]
```diff
@@ -12,258 +11,138 @@
...
-            catch (e) {
+                args = JsonUtils.parse(/** @type {string} */ (params.get('args')));
+            } catch (e) {
                 // Do nothing
             }
         }
         args = Array.isArray(args) ? args : [];
-        return {
+
+        return ({
```
Comment (nibhar) [inline src/lib/RpcServer.es.js]:
```suggestion
        return {
```
Probably a leftover from an annotation.

File: src/lib/RpcServer.es.js [hunk 3]
```diff
@@ -12,258 +11,138 @@
...
             data: {
-                id: parseInt(params.get('id'), 10),
-                command: params.get('command'),
+                id: parseInt(/** @type {string} */ (params.get('id')), 10),
+                command: /** @type {string} */ (params.get('command')),
                 args,
             },
             returnURL: params.get('returnURL'),
-        };
+        });
```
Comment (nibhar) [inline src/lib/RpcServer.es.js]:
```suggestion
        };
```

Comment (nibhar) [inline src/lib/RpcServer.es.js]:
```suggestion
        };
```

File: src/lib/RpcServer.es.js [hunk 4]
```diff
@@ -296,50 +172,75 @@ class State {
...
-            state = new State(message);
+            _state = new RpcState(message);
+            const state = _state;
+
             // Cannot reply to a message that has no source window or return URL
-            if (!('source' in message) && !('returnURL' in message))
-                return;
+            if (!('source' in message) && !('returnURL' in message)) return;
             // Ignore messages without a command
             if (!('command' in state.data)) {
```
Comment (nibhar) [inline src/lib/RpcServer.es.js]: Inline return here, too

**Comment by sisou** [comment]
Rebased because of merge conflicts.

File: src/lib/JsonUtils.js (hunk 1 above)
Comment (sisou) [reply src/lib/JsonUtils.js]: Necessary, as ESLint throws an error.

**Pull request merged**